src/main/java/net/minecraftforge/common/IExtensibleEnum.java

```diff
@@ -59,4 +64,12 @@ static <E extends Enum<E> & StringRepresentable> Codec<E> createCodecForExtensib
                 value -> Either.left(value.getSerializedName())
         );
     }
+
+    /**
+     * Use this instead of {@link ByteBufCodecs#idMapper(IntFunction, ToIntFunction)} for extensible enums
+     * because this will not cache the enum values on construction
+     */
+    static <E extends Enum<E> & StringRepresentable> StreamCodec<ByteBuf, E> createStreamCodecForExtensibleEnum(Supplier<E[]> valuesSupplier) {
+        return ByteBufCodecs.VAR_INT.map(i -> valuesSupplier.get()[i], Enum::ordinal);
+    }
```

Comment:
If the purpose of this method is to be used instead of another, shouldn't we deprecate the old method? Not to remove it, but deprecating it with `@Deprecated` would indicate that the modder should use something else (along with a javadoc that points them to it).

That is unless, of course, this method only applies to String-representable extensible enums, in which case you need to reword your javadoc a little bit.

Reply:
This is for Internal use only. I can specify that in the docs.